src/modules/map/components/openlayers/utils/usePrint.composable.js

```diff
@@ -0,0 +1,99 @@
+import { computed, ref } from 'vue'
+import { useStore } from 'vuex'
+
+import { abortPrintJob, createPrintJob, waitForPrintJobCompletion } from '@/api/print.api.js'
+import { getGenerateQRCodeUrl } from '@/api/qrcode.api.js'
+import log from '@/utils/logging'
+
+const dispatcher = { dispatcher: 'usePrint.composable' }
+
+/** @enum */
+export const PrintStatus = {
+    IDLE: 'IDLE',
+    PRINTING: 'PRINTING',
+    FINISHED_SUCCESSFULLY: 'FINISHED_SUCCESSFULLY',
+    FINISHED_FAILED: 'FINISHED_FAILED',
+}
+
+/**
+ * Gathering of all the logic that will trigger and manage a print request to service-print3
+ *
+ * @param {Map} map
+ */
+export function usePrint(map) {
+    const currentJobReference = ref(null)
+    /** @type {PrintStatus} */
+    const printStatus = ref(PrintStatus.IDLE)
+
+    const store = useStore()
+    const printLayout = computed(() => store.state.print.selectedLayout)
+    const printScale = computed(() => store.state.print.selectedScale)
+    const attributions = computed(() =>
+        store.getters.visibleLayers
+            .map((layer) => layer.attributions)
+            .map((attribution) => attribution.name)
+            .filter((attribution, index, self) => self.indexOf(attribution) === index)
+    )
+    const layersWithLegends = computed(() =>
+        store.getters.visibleLayers.filter((layer) => layer.hasLegend)
+    )
+    const lang = computed(() => store.state.i18n.lang)
+    const projection = computed(() => store.state.map.projection)
+
+    /**
+     * @param {Boolean} printGrid Print the coordinate grid on the finished PDF, true or false
+     * @param {Boolean} printLegend Print all visible layer legend (if they have one) on the map,
+     *   true or false
+     * @returns {Promise<String | null>}
+     */
+    async function print(printGrid = false, printLegend = false) {
+        try {
+            if (currentJobReference.value) {
+                await abortCurrentJob()
+            }
+            printStatus.value = PrintStatus.PRINTING
+            await store.dispatch('generateShortLinks', dispatcher)
```

Comment:
Previously, @ltshb suggested me to call the function directy without saving the result to avoid a dispatch. https://github.com/geoadmin/web-mapviewer/pull/658#discussion_r1505965268

Reply:
That is sound, then I'll move the part of the logic that decide to remove geolocation out of the store, in some function in `shortLink.api.js` so that we can re-use it.
I used the dispatch so that this part of the code wasn't duplicated (removing the geolocation URL param and so on)